src/coffee/directives/api/utils/model-key.coffee

```diff
@@ -135,4 +140,18 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
           return
         #set isInit to true to force redraw after all updates are processed
         child.updateModel model
+
+
+      modelsLength: =>
+        len = 0
+        return len unless @scope.models?
+
+        # isArray .. get length
+        # OR check to see if there is length field which (is essentially cached on the front or backend)
+        if angular.isArray(@scope.models) or @scope.models.length?
+          len = @scope.models.length
```

Comment:
This seemed like the best scenario instead of having to worry about deep comparisons. The reason this would have been a can of worms is due to modelsByRef .. ie if it is modelsByRef then there will not be a watch trigger to refresh.. So I want to avoid special code for that case.